### DISCUSSION TITLE
Is it possible to use a custom literate Haskell preprocessor?

### DISCUSSION THREAD

**Discussion by kamoshi** [haskell/haskell-language-server#3949] [Q&A]
I would like to use [markdown-unlit](https://github.com/sol/markdown-unlit) as the default literate Haskell preprocessor for all standalone `.lhs` files. Is there a way to configure HLS, so that it uses markdown-unlit by default?

For example instead of the regular bird-style I would like to write files like this:

```markdown
Some text here

```haskell
fact :: Integer -> Integer
fact 0 = 1
fact n = n * fact (n-1)
\```

Some text after
```

For this to work, I would probably have to pass `-pgmL markdown-unlit` to GHC, but I'm not sure if there is a way to do it globally, so that it is always enabled when using HLS.

![image](https://github.com/mrcjkb/haskell-tools.nvim/assets/18511281/25c4b8c8-d6da-4c81-b06f-e01c9ce93085)

**Comment by kamoshi** [comment]
I have since found that I can use a `hie.yaml` file to force HLS to pass command to GHC:

```yaml
cradle:
  direct:
    arguments: ["-pgmL", "markdown-unlit"]
```

This is less than ideal in my case however, I would prefer to do this in a way that would work in every directory, regardless of whether there is a `hie.yaml` file or not. Perhaps by passing a setting directly to HLS via `haskell-language-server` command.

**Comment by soulomoon** [comment]
Cradle handles the build tools, build tools handle ghc flag. If I am not mistaken.
